zubhub_frontend/zubhub/src/views/create_activity/createActivityScripts.js

```diff
@@ -0,0 +1,474 @@
+import * as Yup from 'yup';
+import ZubhubAPI from '../../api';
+import { toast } from 'react-toastify';
+import { nanoid } from 'nanoid';
+import {
+  s3 as DO,
+  doConfig,
+  Compress,
+  slugify,
+} from '../../assets/js/utils/scripts';
+
+const API = new ZubhubAPI();
+
+const vars = {
+  activityImages_field_touched: false,
+  inspiringExemplesImages_field_touched: false,
+  upload_in_progress: false,
+};
+
+// ^/file validation functions
+const isImage = (value) => {
+  if (value) {
+    let not_an_image = false;
+    for (let index = 0; index < value.length; index++) {
+      if (value[index].type.split('/')[0] !== 'image') {
+        not_an_image = true;
+      }
+    }
+    return !not_an_image
+  } else {
+    return true;
+  }
+};
+
+const tooManyImages = (value) => {
+  return value? value.length <= 10 : true
+};
+
+const imageSizeTooLarge = (value) => {
+  if (value) {
+    let image_size_too_large = false;
+    for (let index = 0; index < value.length; index++) {
+      if (value[index].size / 1000 > 10240) {
+        image_size_too_large = true;
+      }
+    }
+    return !image_size_too_large
+  } else {
+    return true;
+  }
+};
+
+const allEmpty = (arr) => {
+  let allEmptyValues = true
+  if(arr){
+    arr.forEach(value => {
```

Comment:
Do we need to check for all the images or can we fail if one of the validation fails.

can be done with [array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)